### DISCUSSION TITLE
method registering

### DISCUSSION THREAD

**Discussion by BobMessiaen** [bcb/jsonrpcserver#154] [Q&A]
Dear Beau,

We want to upgrade from version 3.4.2 to 4.2.  But to me it looks like you cannot longer methods from a class (unless they are static methods)?
Is there a way you can you can add a method from a class (not a static method)?

Some sample code to illustrate what we were doing before:

class YYY
  def xxxx(self):
     pass

We registered the function as 

yyy = YYY()
m = "xxxx"
methods.add(getattr(yyy, m), "yyy.{}".format(m))

where the jsonrpc method to access the function was:
yyy.xxxx

How can we achieve the same thing with version 4.2 ?

Thanks in advance for your response!

Bob

**Comment by bcb** [comment]
Methods can only be plain functions, however you can pass any object to the methods using [context](https://jsonrpcserver.readthedocs.io/en/stable/api.html#context) which should give you what you need.

**Reply by iode131** [reply]
This context argument is a life-saver!
In other json-rpc dispatchers I was more familiar with (like json-rpc for instance), I'd use a workaround: adding a partial object instead of the method signature.
But simply doing `dispatch(request, context=self)` works like a charm! Such a neat trick!
Great job. Thx

**Reply by bcb** [reply]
Yeah that’s right, just pass self as the context. If you need to pass more put it all in a dict.
